src/pages/Thanks/ThanksPageOptIn.vue

```diff
@@ -0,0 +1,407 @@
+	const loansResponse = receipt?.items?.values ?? [];
+	const loans = loansResponse
+		.filter(item => item.basketItemType === 'loan_reservation')
+		.map(item => {
+			return {
+				...item.loan,
+				team: item.team,
+			};
+		});
+
+	return loans;
+};
+
+export default {
+	name: 'ThanksPageOptIn',
+	components: {
+		WwwPage,
+		KvButton,
+		OptInSteps,
+		KvExpandable,
+		KvMaterialIcon,
+		CheckoutReceipt,
+		SocialShareV2,
+		BorrowerImage,
+		GuestAccountCreation,
+	},
+	inject: ['apollo', 'cookieStore'],
+	data() {
+		return {
+			openOrderConfirmation: false,
+			openShareModule: false,
+			openCreateAccount: false,
+			mdiChevronDown,
+			userOptedOut: false,
+			isGuest: false,
+			lender: {},
+			loans: [],
+			receipt: null,
+		};
+	},
+	apollo: {
+		preFetch(config, client, { cookieStore, route }) {
+			const transactionId = route.query?.kiva_transaction_id
+				? numeral(route.query?.kiva_transaction_id).value()
+				: null;
+
+			return client.query({
+				query: thanksPageQuery,
+				variables: {
+					checkoutId: transactionId,
+					visitorId: cookieStore.get('uiv') || null,
+				}
+			});
+		}
+	},
+	computed: {
+		selectedLoan() {
+			/**  We should select a loan if we are going to ask for comments for it.
+			* The priority order is:
+			* 1. PFP loan
+			* 2. Partner loan With Team Attribution
+			* 3. Loan with the highest unreservedAmount
+			* loans should be sorted by unreservedAmount.
+			*/
+			const orderedLoans = orderBy(this.loans, ['unreservedAmount'], ['desc']);
+			if (this.hasPfpLoan) {
+				return orderedLoans.find(loan => loan.inPfp);
+			}
+			if (this.hasTeamAttributedPartnerLoan) {
+				return orderedLoans.find(loan => loan?.team?.name);
+			}
+			return orderedLoans[0] || {};
+		},
+		hasPfpLoan() {
+			return this.loans.some(loan => loan.inPfp);
+		},
+		hasTeamAttributedPartnerLoan() {
+			return this.loans.some(loan => loan?.distributionModel === 'fieldPartner' && loan?.team?.name);
+		},
+		borrowerName() {
+			return this.selectedLoan?.name ?? '';
+		},
+		isGuestUsCheckout() {
+			// Is a guest checking out only with US loans?
+			// eslint-disable-next-line no-underscore-dangle
+			return this.isGuest && this.loans.every(loan => loan?.__typename === 'LoanDirect');
+		},
+		hash() {
+			return this.selectedLoan.image?.hash ?? '';
+		},
+		weeksToRepay() {
+			const date = this.selectedLoan?.terms?.expectedPayments?.[0]?.dueToKivaDate ?? null;
+			const today = new Date();
+			if (date) {
+				// Get the number of weeks between the first repayment date (in the future) and now
+				return `${differenceInWeeks(Date.parse(date), today)} weeks`;
+			}
+
+			// Calculating a possible range of weeks between the planned expiration date and a month after
+			const expDate = Date.parse(this.selectedLoan?.plannedExpirationDate);
+			const minDate = differenceInWeeks(expDate, today);
+			const maxDate = differenceInWeeks(addMonths(expDate, 1), today);
+
+			return `${minDate} - ${maxDate} weeks`;
```

Comment:
It's possible that minDate and maxDate will be the same number of weeks, due to rounding, so this should check for that case and only return `${minDate} weeks` if they're the same.